functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/rules/CopyProjectOnTestFailure.java

```diff
@@ -0,0 +1,45 @@
+package org.arquillian.smart.testing.ftest.testbed.rules;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import static java.util.stream.Collectors.toList;
+
+public class CopyProjectOnTestFailure extends TestWatcher {
+
+    TestBed testBed;
+
+    public CopyProjectOnTestFailure(TestBed testBed) {
+        this.testBed = testBed;
+    }
+
+    @Override
+    protected void failed(Throwable e, Description description) {
```

Comment:
That might be useful sometimes, but then maybe we could have it as one rule instead? Where on success it will persist only when you have a property (or maybe passing `true` through the constructor)?